app/src/main/java/com/support/android/designlibdemo/activities/SignPetitionActivity.java

```diff
@@ -35,11 +46,47 @@
     TextView tvPetitionMessage;
 
 
+    public String[] getGeolocation() {
```

Comment:
This is the fix 
    public String[] getGeolocation() {

        double longitude = -122.419;
        double latitude = 37.7749;

        LocationManager lm = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
        Location location = lm.getLastKnownLocation(LocationManager.GPS_PROVIDER);

        if(location!=null){
            longitude = location.getLongitude();
            latitude = location.getLatitude();
        }


        String[] s = new String[2];
        Geocoder geocoder = new Geocoder(this, Locale.getDefault());
        List<Address> addresses = null;


            try {
                addresses = geocoder.getFromLocation(latitude, longitude, 1);
                s[0] = addresses.get(0).getCountryName();
                s[1] = addresses.get(0).getPostalCode();
            } catch (IOException e) {
                e.printStackTrace();
                s[0] = "";
                s[1] = "";

        }
        return s;
    }